src/nvidia-cuda/install.sh

```diff
@@ -0,0 +1,75 @@
+#!/usr/bin/env bash
+
+set -e
+
+INSTALL_CUDNN=${INSTALLCUDNN:-"false"}
+INSTALL_NVTX=${INSTALLNVTX:-"false"}
+CUDA_VERSION=${VERSION:-"latest"}
+CUDNN_VERSION=${CUDNNVERSION:-"latest"}
+
+# NVIDIA's package names include this information
+LATEST_CUDA_VERSION="11.7"
+LATEST_CUDNN_VERSION="8.5.0.96-1"
+if [ "$CUDA_VERSION" = "latest" ]; then CUDA_VERSION="$LATEST_CUDA_VERSION"; fi
+if [ "$CUDNN_VERSION" = "latest" ]; then CUDNN_VERSION="$LATEST_CUDNN_VERSION"; fi
+
+if [ "$(id -u)" -ne 0 ]; then
+    echo -e 'Script must be run as root. Use sudo, su, or add "USER root" to your Dockerfile before running this script.'
+    exit 1
+fi
+
+# Function to run apt-get if needed
+apt_get_update_if_needed() {
+    if [ ! -d "/var/lib/apt/lists" ] || [ "$(ls /var/lib/apt/lists/ | wc -l)" = "0" ]; then
+        echo "Running apt-get update..."
+        apt-get update
```

Comment:
We've actually had a few issues recently (https://github.com/devcontainers/features/pull/98) and are removing this optimization until we're confident it doesn't cause breaks for people.